CMakeLists.txt

```diff
@@ -27,38 +33,49 @@ option(
     ${PROJECT_IS_TOP_LEVEL}
 )
 
+set(CPACK_GENERATOR TGZ)
+
+# includes
 include(GNUInstallDirs)
+include(CPack)
 
 add_library(beman.inplace_vector INTERFACE)
 # [CMAKE.LIBRARY_ALIAS]
 add_library(beman::inplace_vector ALIAS beman.inplace_vector)
 
-target_include_directories(
+target_sources(
+    beman.inplace_vector
+    PUBLIC
+        FILE_SET inplace_vector_public_headers
+        TYPE HEADERS
+        BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/include
+        FILES
+            "${CMAKE_CURRENT_SOURCE_DIR}/include/beman/inplace_vector/inplace_vector.hpp"
+)
+set_target_properties(
+    beman.inplace_vector
+    PROPERTIES VERIFY_INTERFACE_HEADER_SETS ON
+)
+target_compile_features(
     beman.inplace_vector
     INTERFACE
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-        $<INSTALL_INTERFACE:include>
+        "$<$<COMPILE_FEATURES:cxx_std_23>:cxx_std_23>"
+        "$<$<NOT:$<COMPILE_FEATURES:cxx_std_23>>:cxx_std_20>"
```

Comment:
But this exports your min required `cxx_std` on this machine.

Reply:
`inplace_vector` targets a min standard of C++17 I believe.

Reply:
**Note from CI workflow**: Needs C++ 20 as C++17 selectively disables ranges and concepts related functionalities!

Reply:
CK is right here

Reply:
CK is right here regarding the C++ version.

Reply:
Ok that's fine, but what about setting the flags on the target to begin with? I thought we were not setting these properties on build/header targets so that they don't propagate downstream to our consumers? @wusatosi

Reply:
I would recommend to move setting of the feature flags to the unit test target instead of the header library (`INTERFACE` target).

Reply:
Ah I see what's going on here.

Yes we discourage use of `target_compile_features`.

See: https://github.com/bemanproject/exemplar/issues/40#issuecomment-2388995030 , https://github.com/bemanproject/optional26/pull/80

Reply:
I see no problem, and it may be changed later to any decision.

Reply:
@wusatosi may it possible to finish this **PR** right now, I want to continue with the `cmake presets`

Reply:
sorry I cannot include this practice that we knowingly discourage & reject. 

I don't think you've shown that your solution doesn't address the original concern. target_compile_features may lead to ABI break & ODR violation **at link time** as documented in previous reference.

This is also documented elsewhere, see:
https://cgold.readthedocs.io/en/latest/rejected/target-compile-features.html#target-compile-features
https://gitlab.kitware.com/cmake/cmake/-/issues/19136

Reply:
I am sure you are misunderstanding the Problem:
```cmake
# Usage:
#   cmake -G Ninja -S . -B build -D CMAKE_CXX_STANDARD=98 --fresh
#   ninja -C build -v
#   build/exe

cmake_minimum_required(VERSION 3.30...3.31)

project(Test LANGUAGES CXX)

add_library(lib1 lib1.cpp)
target_compile_features(lib1 PUBLIC cxx_std_14)

add_library(lib2 lib2.cpp)
target_compile_features(lib2 PUBLIC cxx_std_11)
# ORIG: target_link_libraries(lib2 PRIVATE lib1)
target_link_libraries(lib2 PUBLIC lib1)

add_executable(exe exe.cpp)
target_link_libraries(exe PRIVATE lib2)
```

As long each static library is build within one build step, you can control it.
### But not if the libraries are build independently and installed without a cmake  config package!

And for a header only library, this ODR-violation can't happen!

```bash
bash-5.2$ CXX=g++ cmake -G Ninja -S . -B build -D CMAKE_CXX_STANDARD=98 --fresh
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (16.0s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/clausklein/Downloads/ODR-violoation/build
bash-5.2$ build/exe
lib1: 201402
lib2: 201402
exe: 201402
bash-5.2$ 
```

Reply:
I am waiting for a pragmatic decision ...

Reply:
Mentioned... waiting for someone to come in and review....

Reply:
see my example https://github.com/bemanproject/inplace_vector/pull/33#discussion_r1861512422

Reply:
Note [Some Professional CMake recommendations](https://discourse.bemanproject.org/t/c-20-modules-in-beman-project-libraies/279/5?u=clausklein)